internal/armada/server/submit_to_log.go

```diff
@@ -173,17 +173,27 @@ func (srv *PulsarSubmitServer) SubmitJobs(ctx context.Context, req *api.JobSubmi
 	}
 
 	// Check if the job can be scheduled on any executor,
-	// to avoid having users wait for a job that may never be scheduled
-	allClusterSchedulingInfo, err := srv.SubmitServer.schedulingInfoRepository.GetClusterSchedulingInfo()
-	if err != nil {
-		err = errors.WithMessage(err, "error getting scheduling info")
-		return nil, err
+	// to avoid having users wait for a job that may never be scheduled.
+	//
+	// We only perform this check for jobs submitted to the legacy scheduler.
+	legacySchedulerJobs := make([]*api.Job, 0, len(apiJobs))
+	for _, apiJob := range apiJobs {
```

Comment:
we could extract this out into a helper function as we have basically the same logic in `submit_from_log`